### ISSUE TITLE
Unrecognized option "breadcrumbs" under "sonata_admin"

### ISSUE THREAD

**Issue by rafrsr** [sonata-project/SonataAdminBundle#4094]
The following configuration does not work in brach 3.x like is described in the [documentation](https://sonata-project.org/bundles/admin/3-x/doc/reference/breadcrumbs.html)

``` yaml
# app/config/config.yml

sonata_admin:
    breadcrumbs:
       # use this to change the default route used to generate the link to the parent object inside a breadcrumb, when in a child admin
       child_admin_route: edit

```

Its posible customize the default route for parent object when in a child admin for only one admin?

parent/show/child/list
parent2/edit/child/list
parent3/edit/child/list

Currently I use `buildBreadcrumbs()` method but this method appear as deprecated, any other way?

**Comment by rafrsr** [comment]
I found another posible solution

ChildAdmin:
````php
public function getBreadcrumbs($action)
    {
        $breadcrumbs = parent::getBreadcrumbs($action);
        /** @var MenuItem $breadcrumb */
        foreach ($breadcrumbs as $breadcrumb) {
            if ($breadcrumb->getName() === (string) $this->getParent()->getSubject()) {
                $breadcrumb->setUri($this->getParent()->generateUrl('show', ['id' => $this->getParent()->getNormalizedIdentifier($this->getParent()->getSubject())]));
            }
        }

        return $breadcrumbs;
    }
```

but this method is marked as deprecated too

**Comment by greg0ire** [comment]
This option is brand new, what version of sonata admin are you using. I don't think it is released yet. cc @cassianotartari

**Comment by greg0ire** [comment]
Regarding deprecated methods : you can write your own implementation of the breadcrumbs builder and inject it in your admin.

**Comment by greg0ire** [comment]
Also, you can use the `config:dump-reference` command to see what configuration is currently supported.

**Comment by cassianotartari** [comment]
Take a look [here](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/BreadcrumbsBuilder.php#L112).

This option is used for cases like this:
When you are at route: ``parent-admin/{id}/child-admin/list`` and you click in the object on the breadcrumb the actual route is ``parent-admin/{id}/edit`` you can change it to ``show``

**Comment by rafrsr** [comment]
> This option is used for cases like this:
When you are at route: parent-admin/{id}/child-admin/list and you click in the object on the breadcrumb the actual route is parent-admin/{id}/edit now you can change it to show, for example

that is my case, but I need use some parents with `show` and others with `edit`

````
parent1/show/child1/list
parent2/edit/child2/list
parent3/edit/child3/list
````

> Also, you can use the config:dump-reference command to see what configuration is currently supported.

I know, only report a missed already documented feature, maybe by mistake.

@greg0ire, In the other hand write my own implementation of the breadcrumbs builder and inject it in the admin, would be a solution but it's not practical, the builder should be written from scratch because the original builder is final.

Can be possible add some configuration or method in parents admin to define the prefered route to generate the breadscrump?

````php
$childAdminRoute = 'show';
````
This configuration for each admin and the `child_admin_route` in the config.yml can be very useful in many cases.

**Comment by greg0ire** [comment]
> @greg0ire, In the other hand write my own implementation of the breadcrumbs builder and inject it in the admin, would be a solution but it's not practical, the builder should be written from scratch because the original builder is final.

That's on purpose, so that people like you speak about their problems and submit PRs to make the breadcrumbs builder more configurable

> Can be possible add some configuration or method in parents admin to define the prefered route to generate the breadscrump?

The admin class is too big (see #3947 ), so no, it is not possible. However, you could introduce a new interface. In the breadcrumbs builder, code could be added like to check for the interface : 

```
if ($admin instance of MyInterfaceWithAMeaningfulName) {
    // do some stuff
}
```

**Comment by phansys** [comment]
I'm closing this issue since configuration node `sonata_admin.breadcrumbs.child_admin_route` was added at #4034 and is available as of version [3.5.0](https://github.com/sonata-project/SonataAdminBundle/releases/tag/3.5.0).

Please, feel free to reopen if you think there is something missing.
Thank you!
